### ISSUE TITLE
Kundera don't allow "persistence.xml" in another place

### ISSUE THREAD

**Issue by msfbrasil** [Impetus/kundera#528]
Hello there!

In order to have an alternative to dynamically load the persistence configurations from an external properties file (issue #526 at "https://github.com/impetus-opensource/Kundera/issues/526"), I've tried to load the "persistence.xml" itself from another place.

I've found a very simple solution provided by "Evgeniy Dorofeev" at "http://stackoverflow.com/questions/13717324/is-it-possible-to-have-persistence-xml-in-a-location-other-than-meta-inf".

The problem is that the solution uses an approach the seems to be incompatible with Kundera loading of "persistence.xml" file.
Solution uses "Thread.currentThread().setContextClassLoader()", and method "findPersistenceMetadatas()" of "PersistenceUnitConfiguration" class tries to load "persistence.xml" through code below:
# 

```
        xmls = this.getClass().getClassLoader().getResources("META-INF/persistence.xml");
```

//            if (xmls == null || !xmls.hasMoreElements())
//            {
//                xmls = Thread.currentThread().getClass().getClassLoader().getResources("META-INF/persistence.xml");
//            }
# 

The first "getResources" invocation fails and, without the commented block that uses current thread's ClassLoader, the EntityManagerFactory initialization fails with error:
# 

18 Fev 2014 11:51:47 [main] ERROR com.impetus.kundera.configure.PersistenceUnitConfiguration.configure - Error occurred during persistence unit configuration, Caused by: .
com.impetus.kundera.utils.InvalidConfigurationException: Could not find any META-INF/persistence.xml file in the classpath
    at com.impetus.kundera.configure.PersistenceUnitConfiguration.findPersistenceMetadatas(PersistenceUnitConfiguration.java:132)
# 

Is there any reason to keep the code above commented? Is there another way to make Kundera load "persistence.xml" file from another location?

Thanks and best regards,
Mauro.

**Comment by mevivs** [comment]
Hi,
please have a look at https://github.com/impetus-opensource/Kundera/issues/453

With 2.10 onwards it is possible to provide external persistence configuration properties. 

```
  xmls = this.getClass().getClassLoader().getResources("META-INF/persistence.xml"); 
```

this is as per JPA. 

You may use Spring PersistenceUnitManager to load these configuration explicitly.

-Vivek

**Comment by msfbrasil** [comment]
Thanks a lot Vivek, and sorry by posting a duplicate.

I've noticed that version 2.10 isn't available at "https://server.oosnmp.net/dist/release/" maven repository yet.

Could you please tell me if and when it will be available?

Thanks again and best regards,
Mauro.

**Comment by mevivs** [comment]
I think you should be able to download it from https://oss.sonatype.org/content/repositories/releases.
given below settings should work for you:

```
<repository>
            <id>sonatype-nexus</id>
            <name>Kundera Public Repository</name>
            <url>https://oss.sonatype.org/content/repositories/releases</url>
            <releases>
            <enabled>true</enabled>
            </releases>
            <snapshots>
            <enabled>false</enabled>
            </snapshots>
          </repository>
```

-Vivek

**Comment by msfbrasil** [comment]
Sorry Vivek, my bad.
I've posted the wrong link.

Just checking it out.

I've noticed that version 2.10 is available at "com/impetus/kundera/2.10/" relative path, but what I couldn't find is "com/impetus/core/kundera-core/2.10" and "com/impetus/client/kundera-cassandra/2.10" which are the libraries that I'm using, in fact.

**Comment by msfbrasil** [comment]
Just a feedback.
I didn't noticed that the "groupid" changed. I was looking at the wrong place.

I've noticed that just today while build from sources. :-p

Best regards,
Mauro.
